gtf/middleware.py

```diff
@@ -32,16 +32,19 @@ def generic_template_finder_view(request, base_path='', extra_context={}):
     for t in possibilities:
         try:
             response = render(request, t, extra_context)
-        except (TemplateDoesNotExist):
+        except (TemplateDoesNotExist, NotADirectoryError):
             continue
         except OSError as e:
             # If there's a directory that matches the template we're looking for,
             # Django will raise a `IsADirectoryError` in `render` instead of a
             # `TemplateDoesNotExist` error. IsADirectoryError was introduced in
             # Python 3 and is a subclass of OSError and its errno corresponds to EISDIR,
-            # so for Python 2 compatibility, OSError is caught instead of IsADirectoryError
+            # but the error is mapped to the Exception NotADirectoryError.
+            # So for Python 2 compatibility, OSError is caught instead of IsADirectoryError
```

Comment:
The comment change here doesn't really make sense to me.
The `IsADirectoryError` is raised when a path ending with a directory is given to `open`, but the `NotADirectoryError` is raised when a path with a file name in it, treated like a directory is given to `open` Ex: `index.html/index.html`.